pkg/controller/service/servicecontroller.go

```diff
@@ -688,7 +688,7 @@ func (s *ServiceController) nodeSyncLoop(period time.Duration) {
 		// round.
 		servicesToUpdate = s.cache.allServices()
 		numServices := len(servicesToUpdate)
-		servicesToUpdate = s.updateLoadBalancerHosts(servicesToUpdate, newHosts)
+		servicesToUpdate = s.updateLoadBalancerHosts(servicesToUpdate, &nodes)
 		glog.Infof("Successfully updated %d out of %d load balancers to direct traffic to the updated set of nodes",
 			numServices-len(servicesToUpdate), numServices)
 
```

Comment:
Same here for the prevHosts and newHosts below:
Does this not need to be refactored to use the new nodeList (instead of hosts)?